crates/core/src/db/queries/metadatas.rs

```diff
@@ -670,6 +683,148 @@ pub fn collection_nfts(conn: &Connection, options: CollectionNftOptions) -> Resu
+                    Expr::tbl(
+                        MeMetadataCollections::Table,
+                        MeMetadataCollections::MetadataAddress,
+                    )
+                    .equals(Metadatas::Table, Metadatas::Address),
+                );
+            },
+            |query| {
+                query.inner_join(
+                    MetadataCollectionKeys::Table,
+                    Expr::tbl(
+                        MetadataCollectionKeys::Table,
+                        MetadataCollectionKeys::MetadataAddress,
+                    )
+                    .equals(Metadatas::Table, Metadatas::Address),
+                );
+            },
+        )
+        .left_join(
+            Listings::Table,
+            Condition::all()
+                .add(
+                    Expr::tbl(Listings::Table, Listings::Metadata)
+                        .equals(Metadatas::Table, Metadatas::Address),
+                )
+                .add(Expr::tbl(Listings::Table, Listings::Seller).equals(
+                    CurrentMetadataOwners::Table,
+                    CurrentMetadataOwners::OwnerAddress,
+                ))
+                .add(Expr::tbl(Listings::Table, Listings::PurchaseId).is_null())
+                .add(Expr::tbl(Listings::Table, Listings::CanceledAt).is_null())
+                .add(
+                    Expr::tbl(Listings::Table, Listings::AuctionHouse)
+                        .ne(pubkeys::OPENSEA_AUCTION_HOUSE.to_string()),
+                )
+                .add(
+                    Expr::tbl(Listings::Table, Listings::Expiry)
+                        .is_null()
+                        .or(Expr::tbl(Listings::Table, Listings::Expiry).gt(current_time)),
+                )
+                .add_option(auction_house.map(|auction_house| {
+                    Expr::col((Listings::Table, Listings::AuctionHouse)).eq(auction_house)
+                })),
+        )
+        .and_where(Expr::col(Metadatas::BurnedAt).is_null())
+        .conditions(
+            is_me_collection,
+            |query| {
+                query.and_where(
+                    Expr::col((
+                        MeMetadataCollections::Table,
+                        MeMetadataCollections::CollectionId,
+                    ))
+                    .eq(collection.clone()),
+                );
+            },
+            |query| {
+                query.and_where(
+                    Expr::col((
+                        MetadataCollectionKeys::Table,
+                        MetadataCollectionKeys::CollectionAddress,
+                    ))
+                    .eq(collection.clone()),
+                );
+            },
+        )
+        .limit(limit)
+        .offset(offset)
+        .order_by((Listings::Table, Listings::Price), Order::Asc)
+        .take();
+
+    let query = query.to_string(PostgresQueryBuilder);
+
+    diesel::sql_query(query)
+        .load(conn)
+        .context("Failed to load nft(s)")
```

Comment:
```suggestion
        .context("Failed to load collection listings")
```

This should be queries/listings and return Listing not NFTs. Still can make use sea query and the conditional to decide where to join in me_collections or metadata_collection_keys since need to filter out listings that don't belong to the collection.

Listings should then have a field called nft that uses the nft_loader to lookup the NFT for the listing.